CRM/Core/Form/EntityFormTrait.php

```diff
@@ -159,19 +163,26 @@ protected function addFormButtons() {
    */
   protected function getEntityDefaults() {
     $defaults = [];
-    if ($this->_action != CRM_Core_Action::DELETE &&
+    if (!$this->isDeleteContext() &&
       $this->getEntityId()
     ) {
       $params = ['id' => $this->getEntityId()];
       $baoName = $this->_BAOName;
       $baoName::retrieve($params, $defaults);
     }
-    foreach ($this->entityFields as $fieldSpec) {
+    foreach ($this->entityFields as &$fieldSpec) {
```

Comment:
I mistrust this construct even though I know it's shorter as we seem to have gotten ourselves in problems with it in the past